framework/platform/bpl/uci/cfg/bpl_cfg.cpp

```diff
@@ -175,33 +175,30 @@ int cfg_get_device_info(BPL_DEVICE_INFO *device_info) { return 0; }
 
 int cfg_get_wifi_params(const char iface[BPL_IFNAME_LEN], struct BPL_WLAN_PARAMS *wlan_params)
 {
-    int retVal = 0;
-    int index  = 0;
-
     if (!iface || !wlan_params) {
+        MAPF_ERR("cfg_get_wifi_params: invalid input: iface = "
+                 << intptr_t(iface) << " wlan_params = " << intptr_t(wlan_params));
         return RETURN_ERR;
     }
 
-    retVal = cfg_get_index_from_interface(iface, &index);
-    if (retVal) {
-        return retVal;
-    }
+    char ifname[BPL_IFNAME_LEN] = {0};
+    utils::copy_string(ifname, iface, BPL_IFNAME_LEN);
 
     // The UCI "disabled" setting is optional, defaults to false if not present
     bool disabled = false;
-    cfg_uci_get_wireless_bool(TYPE_RADIO, index, "disabled", &disabled);
+    cfg_uci_get_wireless_bool(TYPE_RADIO, ifname, "disabled", &disabled);
     wlan_params->enabled = !disabled;
 
     // The UCI "channel" setting is not documented as optional, but for Intel
     // wireless (as probably for other drivers) it is. We do not want to
     // fail when wifi still works fine, so default to "auto" (0) and if
     // can't get the channel from UCI just move on.
     wlan_params->channel = 0;
-    if (!cfg_get_channel(index, &wlan_params->channel)) {
-        MAPF_INFO("UCI: radio" << index << ": channel is not configured assuming auto\n");
+    if (!cfg_get_channel(ifname, &wlan_params->channel)) {
+        MAPF_INFO("UCI: interface " << ifname << ": channel is not configured assuming auto\n");
     }
 
-    return retVal;
+    return RETURN_OK;
```

Comment:
We should not always return success here - depending on the actual operation success/failure.

Reply:
... but it was already like that, so it should be fixed in a separate commit.

And as I explained, I think it actually is OK.